### PR TITLE
Feature/pancancer launcher integration prototype

### PR DESCRIPTION
**Pull request by denis-yuen**
Prototype generation of run json from CWL

Also basic refactoring:
* demonstrate type-safe use of Avro-generated CWL client
* demonstrate type-safe use of Swagger-generated quay.io client

### CODE REVIEW

File: dockstore-client/src/main/java/io/dockstore/client/cli/Client.java [hunk 1]
```diff
@@ -52,16 +38,31 @@
...
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (victoroicr) [inline dockstore-client/src/main/java/io/dockstore/client/cli/Client.java]: Why did this get moved here?

File: README.md [hunk 2]
```diff
@@ -105,6 +105,37 @@ To regenerate the swagger client:
...
+
+ * Client library generated by the [swagger code generator](https://github.com/swagger-api/swagger-codegen)
+ * Is generated based on the quay.io's swagger UI spec
+ * Used by the Dockstore CLI to make http requests to quay.io
+ * If CoreOS changes their API, you will need to regenerate the swagger client.
+ 
+ To regenerate the swagger client:
+ 
+1. Run `echo "{\"library\":\"jersey2\",\"invokerPackage\":\"io.swagger.quay.client\",\"modelPackage\":\"io.swagger.quay.client.model\",\"apiPackage\":\"io.swagger.quay.client.api\"}" > config.json`
+2. Run `java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -i https://quay.io/api/v1/discovery -l java -o <output directory> --library jersey2`. The output directory is where you have dockstore/swagger-java-client/.
```
Comment (victoroicr) [inline README.md]: Should the output directory be swagger-java-quay-client

File: dockstore-client/src/main/java/io/dockstore/client/cli/Client.java (hunk 1 above)
Comment (denis-yuen) [inline dockstore-client/src/main/java/io/dockstore/client/cli/Client.java]: Brain fart in the refactoring I think, moving it back.

File: README.md (hunk 2 above)
Comment (denis-yuen) [inline README.md]: Kind of. These commands are run in the swagger-codegen directory, so it kinda depends where the two directories are in relation to each other, so I left it as `<output directory>` for the developer to fill in.

**Pull request merged**